sdk/data/azcosmos/cosmos_client.go

```diff
@@ -75,10 +75,29 @@ func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (
 	if err != nil {
 		return nil, err
 	}
-	scope, err := createScopeFromEndpoint(endpointUrl)
-	if err != nil {
-		return nil, err
+
+	var scope []string
+
+	if o != nil && o.ClientOptions.Cloud.Services != nil {
+		if svcCfg, ok := o.ClientOptions.Cloud.Services[ServiceName]; ok && svcCfg.Audience != "" {
+			audience := svcCfg.Audience
+			if !strings.HasSuffix(audience, "/.default") {
+				audience = strings.TrimSuffix(audience, "/") + "/.default"
+			}
+			scope = []string{audience}
+			log.Write(azlog.EventRequest, fmt.Sprintf("Using audience from client options: %s", audience))
```

Comment:
This is actually the "scope" now, since you've appended `.default`. We should be accurate with the log message